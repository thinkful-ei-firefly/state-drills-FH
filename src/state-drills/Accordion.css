ul {
	padding: 0;
}

li {
	border: 1px solid #447B93;
	list-style-type: none;
	border-radius: 10px;
}

ul button {
	border: none;
	color: white;
	width: 100%;
	height: 30px;
	background-color: #3E95B6;
}