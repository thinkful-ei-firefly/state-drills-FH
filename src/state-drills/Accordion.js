import React from 'react';
import './Accordion.css'

class Accordion extends React.Component {

	state = {
		itemSelected: ''
	}

	componentDidMount() {} // Called after component mounts

	changeSelectedAccordion(title){
		this.setState({
			itemSelected: title
		});
	}

	renderSections(){
		return this.props.sections.map(section => {
			return <li><button onClick={() => this.changeSelectedAccordion(section.title)}>{section.title}</button>
				{section.title===this.state.itemSelected ? <p> {section.content} </p> : ""}
			</li>;
		});
	}

    render() {
        return (
        	<ul>
        		{this.props.sections && this.renderSections()}
        	</ul>
        	);
    }

    componentDidUpdate() {} // Called after each render

    componentWillUnmount() {}
}

export default Accordion;