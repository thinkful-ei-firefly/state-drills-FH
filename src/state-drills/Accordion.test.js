import React from 'react';
import ReactDOM from 'react-dom';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

describe('<Accordion />', () => {
	const sections = [
	    {
	      title: 'Section 1',
	      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
	    },
	    {
	      title: 'Section 2',
	      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
	    },
	    {
	      title: 'Section 3',
	      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
	    },
	  ]

	it('The component renders an empty li when the sections prop is not supplied', () => {
	    const wrapper = shallow(<Accordion />);
	    expect(toJson(wrapper)).toMatchSnapshot();
	})


	it('The component renders no sections as active by default', () => {
		const wrapper = shallow(<Accordion sections={sections}/>);
  		expect(toJson(wrapper)).toMatchSnapshot();
	})

	it('The component opens a clicked section', () => {
	  const wrapper = shallow(<Accordion sections={sections} />);
	  wrapper.find('button').at(1).simulate('click');
	  expect(toJson(wrapper)).toMatchSnapshot();
	})

	it('The component only opens the last section when multiple sections have been clicked.', () => {
	  const wrapper = shallow(<Accordion sections={sections} />);
	  wrapper.find('button').at(1).simulate('click');
	  wrapper.find('button').at(0).simulate('click');
	  wrapper.find('button').at(2).simulate('click');
	  expect(toJson(wrapper)).toMatchSnapshot();
	})

})