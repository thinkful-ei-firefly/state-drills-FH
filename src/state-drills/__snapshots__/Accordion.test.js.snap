// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> Accordion rendering 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<Accordion /> The component only opens the last section when multiple sections have been clicked. 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
       
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
       
    </p>
  </li>
</ul>
`;

exports[`<Accordion /> The component opens a clicked section 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
       
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
       
    </p>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<Accordion /> The component renders an empty li when the sections prop is not supplied 1`] = `<ul />`;

exports[`<Accordion /> The component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<Accordion /> closes the first tab and opens any clicked tab 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
       
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
       
    </p>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<Accordion /> renders empty given no tabs 1`] = `<ul />`;
